Library/Homebrew/cmd/deps.rb

```diff
@@ -52,45 +63,91 @@ def deps
       all?: ARGV.include?("--all"),
       topo_order?: ARGV.include?("-n"),
       union?: ARGV.include?("--union"),
+      for_each?: ARGV.include?("--for-each"),
     )
+    @with_reqs = ARGV.include?("--include-requirements")
+    @with_optional = ARGV.include?("--include-optional")
+    @with_build = ARGV.include?("--include-build")
+    @skip_recommended = ARGV.include?("--skip-recommended")
+    @annotate = ARGV.include?("--annotate")
```

Comment:
>  Maybe it should fall back to using ARGV.include? everywhere?

Yeh, I agree. 

> My impression is we're trying to move away from scattering global ARGV references everywhere.

I think broadly but at this point it's probably reliant on having a decent Command DSL first.